### ISSUE TITLE
How to set the label to  some files or directories when system boot up 

### ISSUE THREAD

**Issue by xdsb** [smack-team/smack#134]
How to set the label to  some files or directories automatically when system boot up ?  not like using the smack-util tool chsmack "chsmack -a 'attr'  file" in terminal after linux system boot up. Just like setting file_contexts in SELINUX
Thank you.

**Comment by cschaufler** [comment]
SELinux uses a utility to initialize labels at system boot. On a system
that uses systemd this is selinux-autorelabel.service. For a smack
system that uses systemd you will want to follow that example. What
system are you targeting?

**Comment by xdsb** [comment]
smack with systemd, and I want to set smack labels for some files and directories in the rootfs so that the labels would be set at a system boot automatically, just likeing using make_ext4fs with option -S file_contexts to set labels in SElinux when making rootfs img,
could I also set smack labels by this way?

**Comment by cschaufler** [comment]
Unfortunately, the authors of make_ext4fs made it SELinux specific. It
would not be difficult to add support for extended attributes in
general, or just for Smack.

**Comment by xdsb** [comment]
OKay,,thank you for your reply.
So,how to set smack labels to files or directories at system boot up? Is there any solution?

**Comment by cschaufler** [comment]
Are you using systemd? If not, what are you using from system management?

**Comment by xdsb** [comment]
Yeah,we use systemd as system management.

**Comment by cschaufler** [comment]
An example of an implementation of Smack initialization can be found:

https://gerrit.automotivelinux.org/gerrit/gitweb?p=AGL/meta-agl.git;a=tree;f=meta-security/recipes-core/systemd/systemd;h=919d2aa051b7b5c44988c38ada15153e9912f23a;hb=refs/changes/23/18823/4

**Comment by jobol** [comment]
When using yocto, some of the labelling made in post install of packets are managed using `pseudo`. It emulates system calls and takes care of security attributes. There is some magic to then apply it in the image either directly or at first boot. But it does not solves transmutations issues because it does neither know nor emulate the process labels.

In my opinion, you are looking for a tool that based on some predefined rules enforce smack labels on files. A such official Smack's tool doesn't exist, AFAIK.

**Comment by xdsb** [comment]
yeah,@jobol, we just want to set labels to files or directories in the rootfs at system boot or set labels at making rootfs image just like "make_ext4fs -S file_context" in Selinux,
file_contexts:
###########################################
**# Root**
/                   u:object_r:rootfs:s0
**# Data files**
/adb_keys           u:object_r:adb_keys_file:s0
/build\.prop        u:object_r:rootfs:s0
/default\.prop      u:object_r:rootfs:s0
**# Executables**
/init               u:object_r:init_exec:s0
......
###########################################
and then system load predefined rules files that are installed into /etc/smack/accesses.d(such as default-access-domains,...)
